Loop/Managers/DeviceDataManager.swift

```diff
@@ -431,7 +431,12 @@ final class DeviceDataManager {
             log.default("CGMManager:%{public}@ did update with %d values", String(describing: type(of: manager)), values.count)
             loopManager.addGlucoseSamples(values) { result in
                 self.log.default("Asserting current pump data")
-                self.pumpManager?.ensureCurrentPumpData(completion: nil)
+                self.pumpManager?.ensureCurrentPumpData(completion: { (lastSync) in
+                    if let lastSync = lastSync, Date().timeIntervalSince(lastSync) < LoopCoreConstants.inputDataRecencyInterval {
+                        self.log.default("Pump data from %@ is fresh. Triggering loop()", String(describing: type(of: self.pumpManager)))
+                        self.loopManager.loop()
```

Comment:
Unit test for this new code?

Reply:
Generally when you've asked for tests at the top layer managers, I've split it out into a separate object to be testable. I'm not sure that's a good idea here; the logic is fairly simple, and an abstraction that makes things clearer and not more complicated isn't obvious to me. This also feels borderline on the cost/value scale. Are you ok with letting this one slide?

Reply:
Seems like important logic, though, doesn't it?  I hear you that it is difficult to test right now.  Are there other ways to get this under test?

Reply:
I'm not just saying the cost is high. I'm also saying the value is low (not a situation where the test will serve as documentation, simple enough that it's not likely to be misunderstood, not a situation where testing will improve the composability or interface; will actually probably do the opposite, for a piece that is intentionally not reusable). But if you insist, I'll look for ways to do it.

Reply:
So, looking into this more, I think our options are to 1) Refactor DeviceDataManager's collaborators (the other managers) into protocols, so we can create mocks for them and put DeviceDataManager under test. 2) Abstract out some kind of class for this logic to decide whether to loop or not, based on glucose storage success/fail, and pumpdata freshness. A "ToLoopOrNotDecider" class, if you will. Hooking it into glucose storage, and/or results thereof, and pumpMananager freshness or ability to trigger pump refresh, and the ability to trigger a Loop mean a lot more wiring that complicates a user understanding the flow.  All for the purpose of putting just two actual pieces of logic under test; the freshness comparison, and glucose storage success comparison. 

I don't think either option is a net improvement in the readability/simplicity of the code, so I am loathe to do it. Those options are usually good ideas for code that might be reusable or substitutable; but this thin layer of mostly declarative code in the top level Loop Managers isn't like that. Yes, it *is* important logic, and because of that, I think it's important to keep it simple. Unless you strongly object; and I'm guessing since you approved the PR, that you don't, I don't think we should do either of the options above.